packages/hint-sri/src/hint.ts

```diff
@@ -296,17 +362,126 @@ Actual:   ${hashes.join(', ')}`;
         debug(`Validating integrity of: ${evt.resource}`);
 
         await everySeries(validations, async (validation: Function) => {
-            return await validation(evt);
+            return await validation(evt, origin, cacheErrors);
         });
     }
 
+    private async validateElement(evt: ElementFound, origin: string) {
+        if (this.isInCache(evt)) {
+            /*
+             * The item is cached. For the VSCode extension and the
+             * local connector with option 'watch' activated we
+             * should report what we have in the cache after the
+             * first 'scan::end'.
+             */
+            if (this.scanEnded) {
+                const promises = this.getCache(evt).map((error) => {
+                    return this.context.report(error.resource, error.message, error.options);
+                });
+
+                await Promise.all(promises);
+            }
+
+            return;
+        }
+
+        const { resource, element } = evt;
+        let content: NetworkData;
+
+        /*
+         * If the resource is a local file, ignore the analysis.
+         * The sri usually is added on the building process before publish,
+         * so is going to be very common that the sri doesn't exists
+         * for local files.
+         */
+        if (resource.startsWith('file://')) {
+            debug(`Ignoring local resource: ${resource}`);
+
+            return;
+        }
+
+        /*
+         * Requester is not included in webpack bundle for `extension-browser`.
```

Comment:
```suggestion
         * `requestAsync` is not included in webpack bundle for `extension-browser`.
```

Maybe also worth commenting why we're NOT using `Requester` here.